src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java

```diff
@@ -103,18 +103,18 @@ public CreateCaseResponse createCase(String caseDataType, LocalDate dateReceived
             // Start a new camunda workflow (caseUUID is the business key).
             Map<String, String> seedData = new HashMap<>();
             if (Objects.equals(caseDataType, WorkflowConstants.CASE_DATA_TYPE_FOI) && receivedData != null) {
-                seedData.put(WorkflowConstants.CASE_REFERENCE, caseResponse.getReference());
-                seedData.put(WorkflowConstants.DATE_RECEIVED, dateReceived.toString());
-                seedData.put(WorkflowConstants.LAST_UPDATED_BY_USER, userUUID.toString());
                 seedData.put(WorkflowConstants.KIMU_DATE_RECEIVED, receivedData.get(WorkflowConstants.KIMU_DATE_RECEIVED));
                 seedData.put(WorkflowConstants.TOPICS, receivedData.get(WorkflowConstants.TOPICS));
                 seedData.put(WorkflowConstants.REQUEST_QUESTION, receivedData.get(WorkflowConstants.REQUEST_QUESTION));
                 seedData.put(WorkflowConstants.ORIGINAL_CHANNEL, receivedData.get(WorkflowConstants.ORIGINAL_CHANNEL));
-            } else {
-                seedData.put(WorkflowConstants.CASE_REFERENCE, caseResponse.getReference());
-                seedData.put(WorkflowConstants.DATE_RECEIVED, dateReceived.toString());
-                seedData.put(WorkflowConstants.LAST_UPDATED_BY_USER, userUUID.toString());
+            } else if (Objects.equals(caseDataType, WorkflowConstants.CASE_DATA_TYPE_COMP)
+                    && receivedData != null
+                    && Objects.equals(receivedData.get(WorkflowConstants.CHANNEL), WorkflowConstants.CHANNEL_COMP_WEBFORM)) {
```

Comment:
Is it worth extracting this logic chain into a method with a more readable name so it's easier to understand this block? Would also be able to add test for the logic chain easily then too.

Reply:
Good shout simplified it in this area - and added a test